test/test_metrics_writer.py

```diff
@@ -1,7 +1,13 @@
 # -*- coding: utf-8 -*-
+# pylint: disable=redefined-outer-name,no-self-use
```

Comment:
Why do we want to add those? Wouldn't it be better to improve the code and remove this exclusion rules?

Reply:
`redefined-outer-name` is annoying to make work with pytest's fixtures. Basically, the following code:

```
@pytest.fixture
def some_fixture():
  ...

def test_some_feature(some_fixture):
  do_something_with(some_fixture)
```
will cause pytest to call the test function with the computed value of `some_fixture()`, based on the parameter name. This isn't an issue at all - in fact, it's what makes pytest have good test setup ergonomics. But pylint just sees a function argument named the same as a global, and complains. In general, this is a valid complaint, just not here, which is why I'm disabling the warning only for this file.